tests/integration/cache_test.go

```diff
@@ -811,6 +811,40 @@ func TestCacheWithPrefixGet(t *testing.T) {
 
 	ctx := t.Context()
 
+	c, err := cache.New(client, "/foo")
+	if err != nil {
+		t.Fatalf("cache.New: %v", err)
+	}
+	defer c.Close()
+	if err = c.WaitReady(ctx); err != nil {
+		t.Fatalf("cache.WaitReady: %v", err)
+	}
+
+	const testKey = "/foo/a"
+	putResp, err := client.Put(ctx, testKey, "val")
+	if err != nil {
+		t.Fatalf("client.Put(%q, \"val\") failed: %v", testKey, err)
+	}
+	if err := c.WaitForRevision(ctx, putResp.Header.Revision); err != nil {
+		t.Fatalf("cache never caught up to rev %d: %v", putResp.Header.Revision, err)
+	}
+
+	appliedRev := putResp.Header.Revision
+	var latestRev int64
+	for i := 0; i < 5; i++ {
+		r, err := client.Put(ctx, fmt.Sprintf("/bar/x%d", i), fmt.Sprintf("%d", i))
+		if err != nil {
+			t.Fatalf("Put(/bar/x%d): %v", i, err)
+		}
+		latestRev = r.Header.Revision
+	}
+	if latestRev <= appliedRev {
+		t.Fatalf("test setup failed: outside rev (%d) did not advance past applied (%d)", latestRev, appliedRev)
+	}
+	if err := client.RequestProgress(ctx); err != nil {
+		t.Fatalf("RequestProgress: %v", err)
+	}
+
 	tests := []struct {
```

Comment:
Please run those tests also against normal etcd client. So we need to split the `TestCacheWithPrefixGet` into tests that validate that both client and cache have consistent behavior when it comes to progress outside the prefix, but also test that shows that cache reject requests for keys outside prefix.